plugins/amqp_trx_plugin/amqp_trx_plugin.cpp

```diff
@@ -106,6 +106,19 @@ struct amqp_trx_plugin_impl : std::enable_shared_from_this<amqp_trx_plugin_impl>
    }
 
    void on_block_start( uint32_t bn ) {
+      auto* prod_plugin = app().find_plugin<producer_plugin>();
```

Comment:
This is actually a rather expensive operation. Save the pointer in `amqp_trx_plugin_impl`